src/main/java/at/hannibal2/skyhanni/features/gui/customscoreboard/CustomScoreboard.kt

```diff
@@ -0,0 +1,127 @@
+//
+// TODO LIST
+// V2 RELEASE
+//  - Soulflow API
+//  - Bank API (actually maybe not, I like the current design)
+//  - beacon power
+//  - skyblock level
+//  - more bg options (round, blurr, outline)
+//  - countdown events like fishing festival + fiesta when its not on tablist
+//  - CookieAPI https://discord.com/channels/997079228510117908/1162844830360146080/1195695210433351821
+//  - Rng meter display
+//  - shorten time till next mayor https://discord.com/channels/997079228510117908/1162844830360146080/1216440046320746596
+//  - option to hide coins earned
+//  - color options in the purse etc lines
+//  - choose the amount of decimal places in shorten nums
+//
+
+package at.hannibal2.skyhanni.features.gui.customscoreboard
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.events.GuiPositionMovedEvent
+import at.hannibal2.skyhanni.events.GuiRenderEvent
+import at.hannibal2.skyhanni.events.LorenzTickEvent
+import at.hannibal2.skyhanni.utils.ChatUtils
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.RenderUtils.HorizontalAlignment
+import at.hannibal2.skyhanni.utils.RenderUtils.renderStringsAlignedWidth
+import at.hannibal2.skyhanni.utils.TabListData
+import net.minecraftforge.client.GuiIngameForge
+import net.minecraftforge.client.event.RenderGameOverlayEvent
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+typealias ScoreboardElementType = Pair<String, HorizontalAlignment>
+
+class CustomScoreboard {
+
+    private var display = emptyList<ScoreboardElementType>()
+    private var cache = emptyList<ScoreboardElementType>()
+    private val guiName = "Custom Scoreboard"
+
+    @SubscribeEvent
+    fun onRenderOverlay(event: GuiRenderEvent.GuiOverlayRenderEvent) {
+        if (!isEnabled()) return
+        if (display.isEmpty()) return
+
+        RenderBackground().renderBackground()
+
+        if (!TabListData.fullyLoaded && config.displayConfig.cacheScoreboardOnIslandSwitch && cache.isNotEmpty()) {
+            config.position.renderStringsAlignedWidth(cache, posLabel = guiName)
+        } else {
+            config.position.renderStringsAlignedWidth(display, posLabel = guiName)
+
+            if (cache != display) cache = display
+        }
```

Comment:
Cleaner ( I removed the cache setting here, see the suggestion line 75 )
```suggestion
        val render = if (!TabListData.fullyLoaded && config.displayConfig.cacheScoreboardOnIslandSwitch && cache.isNotEmpty()) {
            cache
        } else {
            display
        }
        config.position.renderStringsAlignedWidth(render, posLabel = guiName)
```

Reply:
Here without the cache change, since you didn't like my cache suggestion
```suggestion
        val render = if (!TabListData.fullyLoaded && config.displayConfig.cacheScoreboardOnIslandSwitch && cache.isNotEmpty()) {
            cache
        } else {
            if (cache != display) cache = display
            display
        }
        config.position.renderStringsAlignedWidth(render, posLabel = guiName)
```

Reply:
![image](https://github.com/hannibal002/SkyHanni/assets/45315647/683bbacc-6978-487d-bc13-7c2f2f8da677)
wdym?

Reply:
oh, then it isn't pushed yet

Reply:
it should be

Reply:
the last commits
![image](https://github.com/hannibal002/SkyHanni/assets/85900443/317e0858-b90f-44d6-b2d9-0a19cda722df)